controllers/nutanixcluster_controller.go

```diff
@@ -289,20 +321,94 @@ func (r *NutanixClusterReconciler) reconcileNormal(rctx *nctx.ClusterContext) (r
 
 func (r *NutanixClusterReconciler) reconcileFailureDomains(rctx *nctx.ClusterContext) error {
 	log := ctrl.LoggerFrom(rctx.Context)
-	if len(rctx.NutanixCluster.Spec.FailureDomains) == 0 {
-		log.V(1).Info("no failure domains defined on cluster")
-		conditions.MarkTrue(rctx.NutanixCluster, infrav1.NoFailureDomainsReconciled)
+	log.Info("Reconciling failure domains for cluster")
+
+	// First clear the failure domain related conditions
```

Comment:
why do we need to delete this condition?

Reply:
Refactored.